test/forward_backward_compatibility/check_forward_backward_compatibility.py

```diff
@@ -26,110 +26,94 @@
-    ("aten::softplus_backward.grad_input", datetime.date(2022, 1, 31)),
     ("aten::quantile", datetime.date(2022, 9, 30)),
     ("aten::nanquantile", datetime.date(2022, 9, 30)),
     ("aten::_convolution_double_backward", datetime.date(2022, 3, 31)),
-    ("aten::_scatter_reduce", datetime.date(2022, 1, 31)),
-    ("aten::native_multi_head_self_attention", datetime.date(9999, 1, 1)),
-    ("aten::_native_multi_head_self_attention", datetime.date(9999, 1, 1)),
 ]
 
-ALLOW_LIST_COMPILED = [
-    (
-        re.compile(item[0]),
-        item[1],
-        re.compile(item[2]) if len(item) > 2 else None,
-    ) for item in ALLOW_LIST if item[1] >= datetime.date.today()
+MAX_ALLOWED_PERIOD = datetime.timedelta(days=365)
+
+# WARNING: The operators that are included in this list is extremely unsafe
+# because they would bypass all BC/FC checks for indefinite amount of time.
+# This could cause many severe production issues. Please proceed with extreme caution,
+# if you must add an entry to this list.
+ALLOW_LIST_WITHOUT_DEADLINE = [
+    "c10_experimental",
+    "static",
+    "prim::ModuleDictIndex",
+    "prim::MKLDNNRelu6",
+    "prim::MKLDNNRelu6_",
+    "prim::Concat",
+    # Internal, profiler-specific ops
+    "profiler::_call_end_callbacks_on_jit_fut*",
+    "profiler::_record_function_enter",
+    "aten::_cholesky_helper",
+    "aten::_lstsq_helper",
+    "aten::_syevd_helper",
+    "aten::_linalg_solve_out_helper_",
+    "aten::select_backward",
+    "aten::slice_backward",
+    "aten::diagonal_backward",
+    "aten::rowwise_prune",
+    "aten::adaptive_avg_pool3d_backward",
+    "aten::_embedding_bag_dense_backward",
+    "aten::randperm",
+    "aten::_convolution_nogroup",
+    "aten::miopen_convolution_backward",
+    "aten::miopen_convolution_backward_bias",
+    "aten::miopen_convolution_backward_input",
+    "aten::miopen_convolution_backward_weight",
+    "aten::miopen_convolution_transpose_backward",
+    "aten::miopen_convolution_transpose_backward_input",
+    "aten::miopen_convolution_transpose_backward_weight",
+    "aten::miopen_depthwise_convolution_backward",
+    "aten::miopen_depthwise_convolution_backward_input",
+    "aten::miopen_depthwise_convolution_backward_weight",
+    "prepacked::unpack_prepacked_sizes_conv2d",
+    "prepacked::unpack_prepacked_sizes_linear",
+    "aten::native_multi_head_self_attention",
+    "aten::_native_multi_head_self_attention",
 ]
 
-def allow_listed(schema):
-    for item in ALLOW_LIST_COMPILED:
+def compile_allow_list_with_deadline():
+    output = []
+    for item in ALLOW_LIST_WITH_DEADLINE:
+        deadline = item[1]
+        today = datetime.date.today()
+        if deadline > today and deadline - today < MAX_ALLOWED_PERIOD:
+            output.append((re.compile(item[0]), deadline, re.compile(item[2]) if len(item) > 2 else None))
+        if deadline - today >= MAX_ALLOWED_PERIOD:
+            print("{} will be BC broken for too long. We only allow {} days"
```

Comment:
This is misleading. The operators in question are not necessarily "broken" for this period, they are just scheming schema BC and FC checks.

This warning should also provide guidance about how the test works and what is an appropriate period to set (not just what is allowed). For example, "Operator foo will skip BC and FC schema checks for X days, but only Y days of skipping the checks are permitted. It's recommended that the skip date be chosen only to permit the PR to be merged. Once the PR is merged for a couple days the operator no longer needs to skip these checks."